### PR TITLE
regression 4001 and 4002: add test cases for SM3

### PR DESCRIPTION
**Pull request by jforissier**

### CODE REVIEW

**Comment by jforissier** [comment]
Depends on https://github.com/OP-TEE/optee_os/pull/3473.

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -892,6 +892,43 @@ static const uint8_t hash_data_sha512_out1[] = {
 	0x26, 0xd1, 0xcb, 0x8b, 0x47, 0x45, 0x25, 0xd0
 };
 
+/*
+ * SM3
+ * https://tools.ietf.org/html/draft-sca-cfrg-sm3-02
+ * Appendic A.1
```
[the remark this answers is not in the document]
Comment (jforissier) [reply host/xtest/regression_4000.c]: Will fix

File: host/xtest/regression_4000.c [hunk 2]
```diff
@@ -1165,6 +1206,59 @@ static const uint8_t mac_data_sha512_out1[] = {
 	0xD0, 0x37, 0x2A, 0xFA, 0x2E, 0xBE, 0xEB, 0x3A, /* .7.....: */
 };
 
+/*
+ * SM3 HMAC
+ * GM/T 0042-2015
```
[the remark this answers is not in the document]
Comment (jforissier) [reply host/xtest/regression_4000.c]: No link unfortunately. The document is "Cryptography Industry Standard of the People's Republic of China - GM/T 0042-2015 - Test specification for cryptography and security protocol in tri-element architecture", and is available for a fee (much like ISO standards). If you're lucky a Google search will return a preview of the relevant pages if you look for `GM/T 0042-2015 "D.3"` :wink:

**Comment by jforissier** [comment]
Update

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**